lightningd/pay.c

```diff
@@ -411,6 +411,17 @@ void payment_failed(struct lightningd *ld, const struct htlc_out *hout,
 					 &hout->payment_hash);
 
 #ifdef COMPAT_V052
+	/* Prior to "pay: delete HTLC when we delete payment." we would
+	 * delete a payment on retry, but leave the HTLC. */
+	if (!payment) {
+		log_unusual(hout->key.channel->log,
+			    "No payment for %s:"
+			    " was this an old database?",
+			    type_to_string(ltmp, struct sha256,
```

Comment:
`s/ltmp/tmpctx/`, ha, obviously I did not read things closely enough.

Reply:
Yep, rebase without compile check was a bad idea!  Done now....